src/inspector_socket.cc

```diff
@@ -305,8 +305,8 @@ static ws_decode_result decode_frame_hybi17(const std::vector<char>& buffer,
   if (buffer.size() - kMaskingKeyWidthInBytes < payload_length)
     return FRAME_INCOMPLETE;
 
-  std::vector<char>::const_iterator masking_key = it;
-  std::vector<char>::const_iterator payload = it + kMaskingKeyWidthInBytes;
+  auto masking_key = it;
+  auto payload = it + kMaskingKeyWidthInBytes;
```

Comment:
Is there a guideline that mandates such a change? I believe the code is easier to follow if you know the types. I don't think the types are immediately obvious in this context.

Reply:
For me, I only want to change second one to auto (add `// NOLINT` to the first one).

In the long run, I want to make `https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-auto.html` more clear and actionable.

see https://github.com/nodejs/node/pull/26840/files#r267804403.

Reply:
The current guide says https://github.com/nodejs/node/blob/master/CPP_STYLE_GUIDE.md#using-auto:

> ### Using `auto`
>
> Being explicit about types is usually preferred over using `auto`.
> 
> Use `auto` to avoid type names that are noisy, obvious, or unimportant. When
doing so, keep in mind that explicit types often help with readability and
verifying the correctness of code.

Reply:
I am aware of this. But I want it more clear and enable tools like clang-tidy can check on this.

Reply:
I can drop the commit in this pr. But I do hope we can make clear rules on `auto` in
https://github.com/nodejs/node/pull/26840/files#r267804403

Reply:
`it` is declared with `auto`
https://github.com/nodejs/node/blob/97304269af5f270a8e8de3a065a2b34720e7abab/src/inspector_socket.cc#L242
so semantically `std::vector<char>::const_iterator masking_key = it;` is not only a copy, it's also a cast (from whatever the compiler deduced the type of `it` should be).

So IIUC both LOCs should be in sync.

Reply:
I have drop the commit. @richardlau @refack @eugeneo Can you review this again ?